src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunMachine.java

```diff
@@ -72,21 +88,28 @@ public void install() {
 			}
 		}
 	}
-	
-	public MultiBlock toMultiBlock() {
+
+	private static Material[] convertItemStacksToMaterial(ItemStack[] items) {
 		List<Material> mats = new ArrayList<>();
-		for (ItemStack i: this.getRecipe()) {
+		for (ItemStack i: items) {
```

Comment:
You should use getRecipe() instead because the method could be overridden.

Reply:
Other than that... looks good!

Reply:
Uhm, i dont seem to understand your request. That is a static method, the argument 'items' is passed to it. The former method toMultiBlock() still uses "this.getRecipe()" to generate the actual mutliblock, only the Deprecated Constructors use this method directly to generate the BlockFace-Trigger Argument

Reply:
The static method can of course not be overridden.
The method getRecipe() can be overridden though, using this method would ensure that this case is covered, by accessing the field directly, it could cause redundancies.

Reply:
I am confused. Just to clarify: We are talking about the static Method convertItemStacksToMaterial(ItemStack[] items). This Method is called by toMultiBlock() which uses getRecipe() to get the argument "ItemStack[] items" and passes it to convertItemStacksToMaterial(ItemStack[] items) where you now propose to use getRecipe() instead of the passed "ItemStack[] items" argument. As far as i can see, even If i would implement it there wouldnt be any change in any result, because the 'items' argument is generated by the getRecipe().

Reply:
Oh, sorry I actually did not see that "items" was a parameter, I thought it was a field.
My bad.